perception/autoware_tensorrt_bevdet/include/autoware/tensorrt_bevdet/bevdet_node.hpp

```diff
@@ -0,0 +1,221 @@
+  rclcpp::Subscription<sensor_msgs::msg::CameraInfo>::SharedPtr
+    sub_bl_caminfo_;  ///< Subscriber for back-left camera info
+  rclcpp::Subscription<sensor_msgs::msg::CameraInfo>::SharedPtr
+    sub_br_caminfo_;  ///< Subscriber for back-right camera info
+  std::vector<bool>
+    caminfo_received_;  ///< Flag indicating if camera info has been received for each camera
+  bool camera_info_received_flag_ =
+    false;                    ///< Flag indicating if camera info has been received for all cameras
+  bool initialized_ = false;  /// Flag indicating if img_w_ and img_h_ has been initialized
+
+  // tf listener
+  std::shared_ptr<tf2_ros::TransformListener> tf_listener_{
+    nullptr};                                   ///< TF listener for transforming coordinates
+  std::unique_ptr<tf2_ros::Buffer> tf_buffer_;  ///< Buffer for storing TF transforms
+
+  // Camera parameters;
+  std::vector<Eigen::Matrix3f> cams_intrin;  ///< Intrinsic camera parameters for each camera
+  std::vector<Eigen::Quaternion<float>>
+    cams2ego_rot;  ///< Rotation from camera frame to ego frame for each camera
+  std::vector<Eigen::Translation3f>
+    cams2ego_trans;  ///< Translation from camera frame to ego frame for each camera
+
+  // Subscribers of images for each camera, synchronized
+  message_filters::Subscriber<sensor_msgs::msg::Image>
+    sub_f_img_;  ///< Subscriber for front camera image
+  message_filters::Subscriber<sensor_msgs::msg::Image>
+    sub_fl_img_;  ///< Subscriber for front-left camera image
+  message_filters::Subscriber<sensor_msgs::msg::Image>
+    sub_fr_img_;  ///< Subscriber for front-right camera image
+  message_filters::Subscriber<sensor_msgs::msg::Image>
+    sub_b_img_;  ///< Subscriber for back camera image
+  message_filters::Subscriber<sensor_msgs::msg::Image>
+    sub_bl_img_;  ///< Subscriber for back-left camera image
+  message_filters::Subscriber<sensor_msgs::msg::Image>
+    sub_br_img_;  ///< Subscriber for back-right camera image
+
+  typedef message_filters::sync_policies::ApproximateTime<
+    sensor_msgs::msg::Image, sensor_msgs::msg::Image, sensor_msgs::msg::Image,
+    sensor_msgs::msg::Image, sensor_msgs::msg::Image, sensor_msgs::msg::Image>
+    MySyncPolicy;
+
+  typedef message_filters::Synchronizer<MySyncPolicy> Sync;
+  std::shared_ptr<Sync> sync_;  ///< Synchronizer for synchronizing image callbacks
+
+  // Timer for checking initialization
+  rclcpp::TimerBase::SharedPtr timer_;  ///< Timer for checking initialization
+
+  /**
+   * @brief Starts the subscription to camera info topics for each camera.
+   */
+  void startCameraInfoSubscription();
+
+  /**
+   * @brief Checks if the node has been initialized properly.
+   */
+  void checkInitialization();
+
+  /**
+   * @brief Initializes the object detection model.
+   */
+  void initModel();
+
+  /**
+   * @brief Starts the subscription to image topics for each camera.
+   */
+  void startImageSubscription();
+
+public:
```

Comment:
The callback function `callback` takes six image message pointers as input, but its implementation (not shown) might be quite complex. Consider breaking down this callback into smaller, more manageable functions for better readability, maintainability, and easier debugging.